Segfault when doing role assignment

### Describe the issue

When attempting to do a role assigment we get a segfault as follows:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x123fde0]

goroutine 2533 [running]:
github.com/open-feature/go-sdk-contrib/providers/go-feature-flag/pkg.genericEvaluation[...](0x0, {0x1c47918, 0x40009af950}, {0x186c40a, 0xc}, 0x0, 0x26384)
	github.com/open-feature/go-sdk-contrib/providers/go-feature-flag@v0.1.36/pkg/provider.go:206 +0x70
github.com/open-feature/go-sdk-contrib/providers/go-feature-flag/pkg.(*Provider).BooleanEvaluation(0x1567380?, {0x1c47918?, 0x40009af950?}, {0x186c40a?, 0x24?}, 0x70?, 0x0?)
	github.com/open-feature/go-sdk-contrib/providers/go-feature-flag@v0.1.36/pkg/provider.go:147 +0x6c
github.com/open-feature/go-sdk/openfeature.(*Client).evaluate(0x4000776120, {0x1c47918, 0x40009af950}, {0x186c40a, 0xc}, 0x0, {0x144a7a0, 0x2aba420}, {{0x40005c0a20, 0x24}, ...}, ...)
	github.com/open-feature/go-sdk@v1.11.0/openfeature/client.go:727 +0xbd4
github.com/open-feature/go-sdk/openfeature.(*Client).BooleanValueDetails(0x4000776120, {0x1c47918, 0x40009af950}, {0x186c40a, 0xc}, 0x0, {{0x40005c0a20?, 0x0?}, 0x40009dc240?}, {0x0, ...})
	github.com/open-feature/go-sdk@v1.11.0/openfeature/client.go:397 +0x1a8
github.com/open-feature/go-sdk/openfeature.(*Client).BooleanValue(0x0?, {0x1c47918?, 0x40009af950?}, {0x186c40a?, 0x4000ca6c08?}, 0x0, {{0x40005c0a20?, 0x0?}, 0x40009dc240?}, {0x0?, ...})
	github.com/open-feature/go-sdk@v1.11.0/openfeature/client.go:304 +0x40
github.com/open-feature/go-sdk/openfeature.(*Client).Boolean(0x1c47918?, {0x1c47918?, 0x40009af950?}, {0x186c40a?, 0x150e660?}, 0xc0?, {{0x40005c0a20?, 0x5?}, 0x40009dc240?}, {0x0?, ...})
	github.com/open-feature/go-sdk@v1.11.0/openfeature/client.go:603 +0x34
github.com/stacklok/minder/internal/flags.Bool({0x1c47918, 0x40009af950}, {0x1c64f40, 0x4000776120}, {0x186c40a, 0xc})
	github.com/stacklok/minder/internal/flags/flags.go:52 +0x74
github.com/stacklok/minder/internal/controlplane.(*Server).AssignRole(0x40005a0000, {0x1c47918, 0x40009af950}, 0x1c5f0f8?)
	github.com/stacklok/minder/internal/controlplane/handlers_authz.go:263 +0x138
github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1._PermissionsService_AssignRole_Handler.func1({0x1c47918?, 0x40009af950?}, {0x16d9a00?, 0x400095bb40?})
	github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1/minder_grpc.pb.go:1785 +0xd0
github.com/stacklok/minder/internal/controlplane.ProjectAuthorizationInterceptor({0x1c47918, 0x40009af950}, {0x16d9a00, 0x400095bb40}, 0x400048baa0, 0x40006fd668)
	github.com/stacklok/minder/internal/controlplane/handlers_authz.go:113 +0x324
google.golang.org/grpc.getChainUnaryHandler.func1({0x1c47918, 0x40009af950}, {0x16d9a00, 0x400095bb40})
	google.golang.org/grpc@v1.63.2/server.go:1186 +0xa0
github.com/stacklok/minder/internal/controlplane.EntityContextProjectInterceptor({0x1c47918, 0x40009aee10}, {0x16d9a00, 0x400095bb40}, 0x400048baa0, 0x400095be00)
	github.com/stacklok/minder/internal/controlplane/handlers_authz.go:75 +0x274
google.golang.org/grpc.getChainUnaryHandler.func1({0x1c47918, 0x40009aee10}, {0x16d9a00, 0x400095bb40})
	google.golang.org/grpc@v1.63.2/server.go:1186 +0xa0
github.com/stacklok/minder/internal/controlplane.TokenValidationInterceptor({0x1c47918, 0x40009ae8d0}, {0x16d9a00, 0x400095bb40}, 0x400048baa0, 0x400095bcc0)
	github.com/stacklok/minder/internal/controlplane/handlers_token.go:80 +0x494
google.golang.org/grpc.getChainUnaryHandler.func1({0x1c47918, 0x40009ae8d0}, {0x16d9a00, 0x400095bb40})
	google.golang.org/grpc@v1.63.2/server.go:1186 +0xa0
github.com/stacklok/minder/internal/controlplane.(*Server).StartGRPCServer.Interceptor.func3({0x1c47918, 0x40009ae0f0}, {0x16d9a00, 0x400095bb40}, 0x400048baa0, 0x400095bc80)
	github.com/stacklok/minder/internal/logger/logging_interceptor.go:81 +0x330
google.golang.org/grpc.getChainUnaryHandler.func1({0x1c47918, 0x40009ae0f0}, {0x16d9a00, 0x400095bb40})
	google.golang.org/grpc@v1.63.2/server.go:1186 +0xa0
github.com/stacklok/minder/internal/controlplane.(*Server).StartGRPCServer.SanitizingInterceptor.func2({0x1c47918?, 0x40009ae0f0?}, {0x16d9a00?, 0x400095bb40?}, 0x400048baa0?, 0x40006fd668?)
	github.com/stacklok/minder/internal/util/statuses.go:183 +0x34
google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0x1c47918, 0x40009ae0f0}, {0x16d9a00, 0x400095bb40}, 0x400048baa0, 0x17bc360?)
	google.golang.org/grpc@v1.63.2/server.go:1177 +0x88
github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1._PermissionsService_AssignRole_Handler({0x184c220, 0x40005a0000}, {0x1c47918, 0x40009ae0f0}, 0x40007ce280, 0x400037f180)
	github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1/minder_grpc.pb.go:1787 +0x148
google.golang.org/grpc.(*Server).processUnaryRPC(0x400074c800, {0x1c47918, 0x40009ae000}, {0x1c543a0, 0x40006d0600}, 0x400098e240, 0x40008baea0, 0x2c3f290, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xb58
google.golang.org/grpc.(*Server).handleStream(0x400074c800, {0x1c543a0, 0x40006d0600}, 0x400098e240)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 2532
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x13c
```

Seems to be an issue with the recently included feature flags.

### To Reproduce

```bash
minder project role grant -r editor -s <some subject>
```

### What version are you using?

_No response_

--- a/deployment/helm/templates/deployment.yaml
+++ b/deployment/helm/templates/deployment.yaml
@@ -81,6 +81,8 @@ spec:
             value: "/secrets/github-app/github_app_webhook_secret"
           - name: "MINDER_PROVIDER_GITHUB_APP_FALLBACK_TOKEN_FILE"
             value: "/secrets/github-app/github_app_fallback_token"
+          - name: "MINDER_FLAGS_GO_FEATURE_FILE_PATH"
+            value: "/flags/flags-config.yaml"
           - name: "MINDER_WEBHOOK_CONFIG_WEBHOOK_SECRET_FILE"
             value: "/secrets/app/github_repo_webhook_secret"
           - name: "MINDER_WEBHOOK_CONFIG_PREVIOUS_WEBHOOK_SECRET_FILE"

--- a/internal/flags/flags.go
+++ b/internal/flags/flags.go
@@ -71,6 +71,7 @@ func OpenFeatureProviderFromFlags(ctx context.Context, cfg config.FlagsConfig) {
 		})
 		if err != nil {
 			zerolog.Ctx(ctx).Error().Err(err).Msg("Unable to create GoFeatureFlag provider")
+			flagProvider = nil // Need to explicitly reset the value, see #3259
 		}
 	}
 

--- a/internal/flags/flags_test.go
+++ b/internal/flags/flags_test.go
@@ -1,0 +1,99 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package flags containts utilities for managing feature flags.
+package flags
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/lestrrat-go/jwx/v2/jwt/openid"
+	"github.com/open-feature/go-sdk/openfeature"
+	"github.com/stacklok/minder/internal/auth"
+	"github.com/stacklok/minder/internal/engine"
+
+	config "github.com/stacklok/minder/internal/config/server"
+)
+
+func TestOpenFeatureProviderFromFlags(t *testing.T) {
+	t.Parallel()
+
+	testFile := filepath.Join(t.TempDir(), "testfile.yaml")
+	tempFile, err := os.Create(testFile)
+	if err != nil {
+		t.Fatalf("failed to create temp file: %v", err)
+	}
+	t.Cleanup(func() { tempFile.Close() })
+	configFile := `
+idp_resolver:
+  variations:
+    FlagOn: true
+    FlagOff: false
+  defaultRule:
+    variation: FlagOn
+`
+	if _, err := io.WriteString(tempFile, configFile); err != nil {
+		t.Fatalf("failed to write to temp file: %v", err)
+	}
+
+	tests := []struct {
+		name         string
+		cfg          config.FlagsConfig
+		expectedFlag bool
+	}{{
+		name: "No Config",
+		cfg:  config.FlagsConfig{},
+	}, {
+		name: "No File",
+		cfg: config.FlagsConfig{
+			GoFeature: config.GoFeatureConfig{
+				FilePath: "non-existent-file",
+			},
+		},
+	}, {
+		name: "File exists",
+		cfg: config.FlagsConfig{
+			GoFeature: config.GoFeatureConfig{
+				FilePath: testFile,
+			},
+		},
+		expectedFlag: true,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			OpenFeatureProviderFromFlags(ctx, tt.cfg)
+
+			client := openfeature.NewClient("test")
+			userJWT := openid.New()
+			userJWT.Set("sub", "user-1")
+			ctx = auth.WithAuthTokenContext(ctx, userJWT)
+			ctx = engine.WithEntityContext(ctx, &engine.EntityContext{
+				Project:  engine.Project{uuid.New()},
+				Provider: engine.Provider{"testing"},
+			})
+
+			flagResult := Bool(ctx, client, IDPResolver)
+			if flagResult != tt.expectedFlag {
+				t.Errorf("expected %v, got %v", tt.expectedFlag, flagResult)
+			}
+		})
+	}
+}

--- a/internal/flags/flags_test.go
+++ b/internal/flags/flags_test.go
@@ -26,21 +26,21 @@ import (
 	"github.com/google/uuid"
 	"github.com/lestrrat-go/jwx/v2/jwt/openid"
 	"github.com/open-feature/go-sdk/openfeature"
-	"github.com/stacklok/minder/internal/auth"
-	"github.com/stacklok/minder/internal/engine"
 
+	"github.com/stacklok/minder/internal/auth"
 	config "github.com/stacklok/minder/internal/config/server"
+	"github.com/stacklok/minder/internal/engine"
 )
 
 func TestOpenFeatureProviderFromFlags(t *testing.T) {
 	t.Parallel()
 
-	testFile := filepath.Join(t.TempDir(), "testfile.yaml")
+	testFile := filepath.Clean(filepath.Join(t.TempDir(), "testfile.yaml"))
 	tempFile, err := os.Create(testFile)
 	if err != nil {
 		t.Fatalf("failed to create temp file: %v", err)
 	}
-	t.Cleanup(func() { tempFile.Close() })
+	t.Cleanup(func() { _ = tempFile.Close() })
 	configFile := `
 idp_resolver:
   variations:
@@ -78,16 +78,19 @@ idp_resolver:
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
 			ctx := context.Background()
 			OpenFeatureProviderFromFlags(ctx, tt.cfg)
 
 			client := openfeature.NewClient("test")
 			userJWT := openid.New()
-			userJWT.Set("sub", "user-1")
+			if err := userJWT.Set("sub", "user-1"); err != nil {
+				t.Fatalf("failed to set sub claim: %v", err)
+			}
 			ctx = auth.WithAuthTokenContext(ctx, userJWT)
 			ctx = engine.WithEntityContext(ctx, &engine.EntityContext{
-				Project:  engine.Project{uuid.New()},
-				Provider: engine.Provider{"testing"},
+				Project:  engine.Project{ID: uuid.New()},
+				Provider: engine.Provider{Name: "testing"},
 			})
 
 			flagResult := Bool(ctx, client, IDPResolver)

